### DISCUSSION TITLE
Oracle database

### DISCUSSION THREAD

**Discussion by mtlc1402** [PiranhaCMS/piranha.core#1413] [Q&A] [answered]
I'd like to use Piranha with Oracle. There's Oracle EF core provider for .Net core. What should I do next to make it work? I'm new and open to learn. Really appreciate for your help.

**Comment by tidyui** [comment, marked as answer]
Hi there! Creating a EF Db implementation is merely about adding an implementation of `Piranha.IDb` deriving from `Piranha.Db<T>`, a `DbFactory` for running migrations, and then creating the actual migrations for Oracle. See the `SqLite` implementation for reference:

* https://github.com/PiranhaCMS/piranha.core/blob/master/data/Piranha.Data.EF.SQLite/SQLiteDb.cs
* https://github.com/PiranhaCMS/piranha.core/blob/master/data/Piranha.Data.EF.SQLite/DbFactory.cs

As we don't have the means to test **or** maintain a provider for Oracle we're not interested in adding it into our repository, but if you create one and publish it on NuGet we'll make sure to promote it on our website and documentation!

Best regards

**Comment by mtlc1402** [comment]
@tidyui Thank you!!!
